qrisk/stats.py

```diff
@@ -164,69 +140,84 @@ def max_drawdown(returns):
     if returns.size < 1:
         return np.nan
 
-    df_cum_rets = cum_returns(returns, starting_value=100)
-    cum_max_return = df_cum_rets.cummax()
+    current_max = returns.copy()
+    for i in range(1, current_max.size):
```

Comment:
we changed the formula other than the cummax though, I don't think it was the cummax that was the problem. This look will be very slow for large inputs.